air/src/aux_table/bitwise/mod.rs

```diff
@@ -17,19 +18,28 @@ pub mod tests;
 // ================================================================================================
 
 /// The number of transition constraints on the bitwise co-processor.
-pub const NUM_CONSTRAINTS: usize = 14;
-/// The degrees of constraints on the bitwise co-processor. The degree of all
+pub const NUM_CONSTRAINTS: usize = 18;
+/// The degrees of constraints on the bitwise co-processor. The degree of all bitwise
 /// constraints is increased by 4 due to the co-processor selector flag from the auxiliary table
-/// (degree 2) and the selector flag specifying the bitwise operation (degree 2).
+/// (degree 2) and the internal selector flag specifying the bitwise operation (degree 2). The
+/// degree of internal selectors is increased by 2 due to co-processor selector flag from the
+/// Auxiliary Table.
 pub const CONSTRAINT_DEGREES: [usize; NUM_CONSTRAINTS] = [
+    4, 4, // Internal selector flags must be binary.
+    4, 4, // Internal selector flags should remain the same throughout the cycle.
     6, 6, 6, 6, 6, 6, 6, 6, // Input decomposition values should be binary.
     6, 6, // Enforce correct initial values of a and b columns.
     6, 6, // Enforce correct aggregation of a and b columns during transitions.
     7, 7, // Ensure correct output aggregation
 ];
 
-/// Index of CONSTRAINT_DEGREES array after which all constraints use periodic columns.
-const PERIODIC_CONSTRAINTS_START: usize = 8;
+/// Index of CONSTRAINT_DEGREES array after which all internal selector constraints
+/// use periodic columns.
+const INTERNAL_SELECTOR_PERIODIC_CONSTRAINT_STARTS: usize = 2;
+/// Index of CONSTRAINT_DEGREES array after which all internal selector constraints end.
+const INTERNAL_SELECTOR_CONSTRAINT_ENDS: usize = 4;
+/// Index of CONSTRAINT_DEGREES array after which all bitwise constraints use periodic columns.
+const PERIODIC_CONSTRAINTS_START: usize = 12;
```

Comment:
nit: I think at this point it might be more straightforward if we switch to explicitly defining the constraint degrees explicitly inside `get_transition_constraint_degrees` the way that it is done in the [hasher air module](https://github.com/maticnetwork/miden/blob/next/air/src/aux_table/hasher/mod.rs). However what you've done here is clear enough that I don't feel strongly, so feel free to disagree :)

Reply:
As it's already been done in PR #281, it makes sense if we define the constraint explicitly.